test/index.test.js

```diff
@@ -42,10 +42,7 @@ test('returns non-chunked payload', (t) => {
     t.error(err)
     t.equal(res.statusCode, 200)
     t.equal(res.statusMessage, 'Super')
-    t.ok(res.headers.date)
     t.strictSame(res.headers, {
-      date: res.headers.date,
-      connection: 'keep-alive',
```

Comment:
why are date and connection gone?

Reply:
Previously, you read these options from `_header` which is not documented in the `ServerResponse`
(also Bun not export this attribute)

Reply:
Now we read header from `.getHeader(name)`

Reply:
This is a breaking change. Probably would result in delayed approval of this PR. Can you implement it not backwards breaking?

Reply:
I think not because node did it in a hidden way and bun doesn't add these headers at all

Reply:
I can wait, no problem)

Reply:
It could be also result in a rejection of your PR, as bun is incompatible with nodejs despite @Jarred-Sumner 's claim to be a drop in replacement. 

Please reconsider.

Reply:
So I think I can fix these headers in Bun, but you are also using undocumented options, don't you think that is not quite the right way
I think users shouldnt read and interact with them at all

Reply:
We are devs, we should always agree on a reasonable solution.

But for a informed decision, it is not enough to write "undocumented options". Can you please elaborate what options we are using, which are undocumented etc.?

But on the other hand, we have basically an api contract to fulfill. If you remove headers, we will break the expectations. 

Thats why I wrote, that if you could implement it in a non-breaking way. Because we will have soon fastify v5, where we can introduce breaking changes. So we are in the perfect time window to discuss if those headers should be added or not. And if not, we can remove them in few months, when we release fastify v5.

But you should not be sad, if removing the additional headers is not accepted.

Reply:
I totally agree with you when I said "undocumented parameters" I meant "ServerResponse._header".

https://nodejs.org/api/http.html#class-httpoutgoingmessage (no word about `_header`)